core/app/models/spree/order_update_attributes.rb

```diff
@@ -0,0 +1,36 @@
+module Spree
+  class OrderUpdateAttributes
+    def initialize(order, attributes, request_env: nil)
+      @order = order
+      @attributes = attributes.dup
+      @payments_attributes = @attributes.delete(:payments_attributes) || []
+      @request_env = request_env
+    end
+
+    def apply
+      assign_payments_attributes
+      assign_order_attributes
+
+      if order.save
+        order.set_shipments_cost if order.shipments.any?
+        true
+      else
+        false
+      end
+    end
+
+    private
+
+    attr_reader :attributes, :payments_attributes, :order
+
+    def assign_order_attributes
+      order.attributes = attributes
```

Comment:
In general I also prefer `assign_attributes` over `attributes=` since I find the latter to be a little misleading, but I'm not sure it'd be worth the time it would take for john to retype it. :) (Personally I like to defer to the author of the code on minor stylistic stuff.)